hls4ml/model/hls_layers.py

```diff
@@ -1384,6 +1384,9 @@ def initialize(self):
                 self.set_attr('implementation', 'latency')
             else:
                 self.set_attr('implementation', self.model.config.get_strategy(self).lower())
+            
+            if self.model.config.get_config_value('IOType') == 'io_parallel':
+                assert len(self.get_input_variable().shape) == 1, 'Softmax with io_parallel strategy cannot be used on multidimensional tensors.'
```

Comment:
Are you happy with this here, or want to move it to the backend for easier integration with the backend PR?

Reply:
The limits of the `io_stream` implementation (the `axis` restriction) is made as an assert in C++/HLS, I was thinking the same should be done here, but there is no way to know the shape info at that stage in the C++. For the integration with backend PR this line is simply placed in the backend initializer for the layer, it is not difficult to follow up.